src/components/encoding-pane/index.tsx

```diff
@@ -6,26 +6,38 @@ import {Channel} from 'vega-lite/build/src/channel';
 import * as styles from './encoding-pane.scss';
 
 import {SHORT_WILDCARD} from 'compassql/build/src/wildcard';
+import {CompassAsyncAction, compassRecomendationsLoad} from '../../actions/compass';
 import {ActionHandler} from '../../actions/index';
 import {createDispatchHandler} from '../../actions/redux-action';
 import {SHELF_CLEAR, ShelfAction} from '../../actions/shelf';
 import {ShelfUnitSpec, State} from '../../models';
 import {EncodingShelf} from './encoding-shelf';
 import {MarkPicker} from './mark-picker';
 
-interface EncodingPanelProps extends ActionHandler<ShelfAction> {
+interface EncodingPanelProps extends ActionHandler<ShelfAction | CompassAsyncAction> {
   spec: ShelfUnitSpec;
 
   specPreview: ShelfUnitSpec;
 }
 
+
 class EncodingPanelBase extends React.PureComponent<EncodingPanelProps, {}> {
   constructor(props: EncodingPanelProps) {
     super(props);
 
     // Bind - https://facebook.github.io/react/docs/handling-events.html
     this.onClear = this.onClear.bind(this);
   }
+
+  public componentDidMount() {
+    this.props.handleAction(compassRecomendationsLoad());
+  }
+
+  public componentDidUpdate() {
+    this.props.handleAction(compassRecomendationsLoad());
```

Comment:
we can research more. this is the simplest mechanism - but yes could result in unnecessary renders. I was asking the same question when implementing this. The most common / best would be to keep it all in the action creators - but since this request to get recommendations depends so much on the different pieces of the `shelf` - it seemed a waste to try to hook all those action creators to additionally call the update to recommend()

Putting it in the reducer would mean the reducer has side effects (which is frowned upon), but a similar question (https://github.com/reactjs/reselect/issues/142) points to researching [Redux Loop](https://github.com/redux-loop/redux-loop) a bit to see if there is a sane way to do it in the reducer.

An easy alternative would be to just add a bit of checking to this `componentDidUpdate` to only call if the shelf changes.